internal/db/models/instance.go

```diff
@@ -38,16 +40,69 @@ const (
 // Instance represents a compute instance in the system
 type Instance struct {
 	gorm.Model
-	JobID      uint           `json:"job_id" gorm:"not null;index"` // ID from the jobs table
-	ProviderID ProviderID     `json:"provider_id" gorm:"not null"`
-	Name       string         `json:"name" gorm:"not null;index"`
-	PublicIP   string         `json:"public_ip" gorm:"varchar(100)"`
-	Region     string         `json:"region" gorm:"varchar(255)"`
-	Size       string         `json:"size" gorm:"varchar(255)"`
-	Image      string         `json:"image" gorm:"varchar(255)"`
-	Tags       pq.StringArray `json:"tags" gorm:"type:text[]"`
-	Status     InstanceStatus `json:"status" gorm:"index"`
-	CreatedAt  time.Time      `json:"created_at" gorm:"index"`
+	JobID         uint           `json:"job_id" gorm:"not null;index"`
+	ProviderID    string         `json:"provider_id" gorm:"not null"`
+	Name          string         `json:"name" gorm:"not null;index"`
+	PublicIP      string         `json:"public_ip" gorm:"varchar(100)"`
+	Region        string         `json:"region" gorm:"varchar(255)"`
+	Size          string         `json:"size" gorm:"varchar(255)"`
+	Image         string         `json:"image" gorm:"varchar(255)"`
+	Tags          pq.StringArray `json:"tags" gorm:"type:text[]"`
+	Status        InstanceStatus `json:"status" gorm:"index"`
+	Volumes       pq.StringArray `json:"volumes" gorm:"type:text[]"`
+	VolumeDetails VolumeDetails  `json:"volume_details" gorm:"type:jsonb"`
+	CreatedAt     time.Time      `json:"created_at" gorm:"index"`
+}
+
+// VolumeDetail contains information about an attached volume
+type VolumeDetail struct {
+	ID         string `json:"id"`          // Volume ID
+	Name       string `json:"name"`        // Volume name
+	SizeGB     int    `json:"size_gb"`     // Size in gigabytes
+	MountPoint string `json:"mount_point"` // Where the volume is mounted
+}
+
+// VolumeDetails type for handling JSONB in database
+type VolumeDetails []VolumeDetail
+
+// Scan implements the sql.Scanner interface
+func (vd *VolumeDetails) Scan(value interface{}) error {
```

Comment:
two points here:
1. It will be more clear if we keep volume stuff in a separate file just like other models
2. I don't see the requirement of defining a Scan method here. This is usually used for special types in the DB where database does not support and we need conversion, but here I see we have simple types. So the regular gorm methods should be enough for it.